.github/workflows/gui-checks.yml

```diff
@@ -0,0 +1,197 @@
+name: GUI Checks
+on: workflow_call
+
+# Cancel in-progress workflows if a new one is started
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-gui-checks
+  cancel-in-progress: true
+
+permissions:
+  contents: read # Read-only access to repository contents
+  issues: write # Write access to issues
+  pull-requests: write # Write access to pull requests
+  statuses: write # Write access to commit statuses
+  checks: write
+
+jobs:
+  lint:
+    name: 👮 Lint GUI
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.ref }}
+
+      - name: 📦 Setup pnpm
+        uses: pnpm/action-setup@v4
+
+      - uses: actions/setup-node@v4
+        name: ⎔ Setup Node
+        with:
+          node-version: 20
+          cache: "pnpm"
+
+      - uses: actions/cache/restore@v4
+        name: Download cache
+        id: cache
+        with:
+          path: |
+            **/.eslintcache
+            node_modules/.cache/prettier
+          key: ${{ runner.os }}-gui-${{ github.run_id }}
+          restore-keys: |
+            ${{ runner.os }}-gui
+
+      - if: startsWith(runner.name, 'GitHub Actions') || startsWith(runner.name, 'Hosted Agent')
+        name: Installing wasm-pack
+        uses: jetli/wasm-pack-action@v0.4.0
+        with:
+          version: v0.12.1
+
+      - name: 📦 Install dependencies
+        run: pnpm install --frozen-lockfile
+
+      - name: 📝 Prettier
+        id: prettier
+        continue-on-error: true
+        run: pnpm run ci:prettier
+
+      # Next Tasks are depend on Typecheck, because we build libraries at this stage
+      - name: 🧠 Typecheck
+        id: typecheck
+        continue-on-error: true
+        run: pnpm run ci:typecheck
+
+      - name: 🧹 Lint
+        id: lint
+        continue-on-error: true
+        run: pnpm run ci:lint
+
+      - name: 🧪 Unit Tests
+        id: unit-tests
+        continue-on-error: true
+        run: pnpm run ci:test
```

Comment:
This might be fine, but just be aware that those jobs used to be running in parallel. Since most of those jobs are actually long-running single-threaded node processes, there wasn't much issue with doing that. With this new setup, they will all be serialized.

Reply:
Yeah, they run in serial now, but lint and unit testing depend on libs to be compiled atm, and lint/prettier have caches (which speed things up), also, one failed task won't cancel the whole workflow.

Also making them fully parallel is not necessary because it's not the longest task out there